diff --git a/kivy/input/providers/androidjoystick.py b/kivy/input/providers/androidjoystick.py
index caf0c06..5d4b3f0 100644
--- a/kivy/input/providers/androidjoystick.py
+++ b/kivy/input/providers/androidjoystick.py
@@ -12,6 +12,7 @@ information, please refer to
 __all__ = ('AndroidMotionEventProvider', )
 
 import os
+# from fnmatch import fnmatch
 
 try:
     import android  # NOQA
@@ -49,13 +50,19 @@ class AndroidMotionEventProvider(MotionEventProvider):
         self.window = None
 
     def create_joystick(self, index):
-        Logger.info('Android: create joystick <%d>' % index)
         js = pygame.joystick.Joystick(index)
         js.init()
         if js.get_numbuttons() == 0:
             Logger.info('Android: discard joystick <%d> cause no button' %
                         index)
+            js.quit()
             return
+#         if not fnmatch(js.get_name(), "Android*touch*"):
+#             Logger.info('Android: discard joystick <%d> cause name not matched' %
+#                         index)
+#             js.quit()
+#             return
+        Logger.info('Android: create joystick <%d>' % index)
         self.joysticks.append(js)
 
     def start(self):
@@ -65,6 +72,7 @@ class AndroidMotionEventProvider(MotionEventProvider):
             self.create_joystick(i)
 
     def stop(self):
+        for joy in self.joysticks: joy.quit()
         self.joysticks = []
 
     def update(self, dispatch_fn):
